### PR TITLE
Docs: Updates for Deprecation of CNI network-plugin Flag

### LINKED ISSUE #26531: Docs: Remove Deprecated network-plugin Kubelet Flag
From the [docs](https://docs.cilium.io/en/stable/network/kubernetes/requirements/):
>[CNI](https://docs.cilium.io/en/stable/glossary/#term-CNI) - Container Network Interface is the plugin layer used by Kubernetes to delegate networking configuration. CNI must be enabled in your Kubernetes cluster in order to install Cilium. This is done by passing --network-plugin=cni to kubelet on all nodes.

The `--network-plugin` flag was [removed](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) in k8s v1.24 and should be removed or updated accordinly in the docs.

### PR DESCRIPTION
**Pull request by danehans**
- Updates kubelet command-line parameters in K8s requirements section.
- Updates install steps for deprecated CLI flag in minikube install section.
- Since the minikube install docs state `minikube ≥ v1.28.0`, notes specific to older minikube versions were removed.

Fixes: #26531

### CODE REVIEW

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joamaki) [inline Documentation/gettingstarted/k8s-install-default.rst:144]: Would it make sense to flip this so that we'd say in the note that for v1.12.0 and earlier the option is `--network-plugin-cni --cni=false` and Cilium is installed manually? Or is that already so ancient that it's not worth mentioning?

Comment (danehans) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: I can make this change if needed. IMHO it comes down to whether docs should be provided for an unsupported minikube version.

Comment (danehans) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: @joamaki do you have an opinion on my ^ comment?

**Review by learnitall — APPROVED** [review]
This looks great, thank you!

As a follow-up to this, it would be great if we could add information on how to install a specific version of Cilium into minikube using the manifest option for the `--cni` parameter. I think the workflow would involve rendering a manifest using `helm template` and then passing the file path of the manifest to `--cni`. @danehans if you have time and would like to tackle this as part of your PR then please feel free, otherwise I can just make a follow-up issue.

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (learnitall) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: I think it makes sense to leave the old version out. Eventually, we'd have to remove the old instructions anyway. If users need to see the old instructions, they can go back in time to view the old installation instructions using the version selector in the docs.

Comment (learnitall) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: I think it makes sense to leave the old version out. Eventually, we'd have to remove the old instructions anyway. If users need to see the old instructions, they can go back in time to view the old installation instructions using the version selector in the docs.

Comment (danehans) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: ^ is exactly what I was thinking.

**Comment by danehans** [comment]
> if you have time and would like to tackle this as part of your PR then please feel free, otherwise I can just make a follow-up issue.

@learnitall thanks for the review. I believe ^ is a separate issue from the one this PR is fixing. Feel free to create an issue and assign me.

**Comment by learnitall** [comment]
Sounds good @danehans, will do!

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (julianwiedmann) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: holding off the merge until this discussion is resolved, @joamaki please ping / re-label when you're good :)

Comment (danehans) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: @joamaki when you have a moment, do you mind commenting to unblock this PR?

Comment (danehans) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: @julianwiedmann do you have any suggestions on how to move this PR forward?

**Comment by learnitall** [comment]
@joamaki, when you have a moment, do you mind taking another look so we can resolve the feedback you brought up?

**Comment by danehans** [comment]
@learnitall @julianwiedmann is it possible to [dismiss the PR review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review) from @joamaki to get this unblocked?

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joamaki) [reply Documentation/gettingstarted/k8s-install-default.rst:144]: Makes sense to me!

**Review by joamaki — APPROVED** [review]

**Pull request merged**
